apps/meteor/client/sidebar/header/MatrixFederationSearch/useInfiniteFederationSearchPublicRooms.tsx

```diff
@@ -0,0 +1,11 @@
+import { useEndpoint } from '@rocket.chat/ui-contexts';
+import { useInfiniteQuery } from '@tanstack/react-query';
+
+export const useInfiniteFederationSearchPublicRooms = (serverName: string, roomName?: string, count?: number) => {
+	const fetchRoomList = useEndpoint('GET', '/v1/federation/searchPublicRooms');
+	return useInfiniteQuery(
+		['federation/searchPublicRooms', serverName, roomName, count],
```

Comment:
I suggest adding `refetchOnWindowFocus: false` to this query and to others that does not need to be refetched